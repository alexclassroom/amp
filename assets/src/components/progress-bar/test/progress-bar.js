/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import { create } from 'react-test-renderer';

/**
 * Internal dependencies
 */
import { ProgressBar } from '../index';

describe('ProgressBar', () => {
	it('matches the snapshot', () => {
		const wrapper = create(<ProgressBar value={33} />);

		expect(wrapper.toJSON()).toMatchSnapshot();
	});

	it('renders a progress bar', () => {
		const { container } = render(<ProgressBar value={33} />);

		expect(
			container.querySelector('.progress-bar[role="progressbar"]')
		).not.toBeNull();
		expect(
			container.querySelector('.progress-bar[aria-valuemin="0"]')
		).not.toBeNull();
		expect(
			container.querySelector('.progress-bar[aria-valuemax="100"]')
		).not.toBeNull();
		expect(
			container.querySelector('.progress-bar[aria-valuenow="33"]')
		).not.toBeNull();
	});

	it('the bar is shifted correctly', () => {
		const { container } = render(<ProgressBar value={75} />);

		expect(
			container.querySelector('.progress-bar[aria-valuenow="75"]')
		).not.toBeNull();
		expect(
			container.querySelector('.progress-bar__indicator')
		).not.toBeNull();
		expect(
			container.querySelector('.progress-bar__indicator').style.transform
		).toBe('translateX(-25%)');
	});

	it('does not allow the bar to be completely out of view for low values', () => {
		const { container } = render(<ProgressBar value={1} />);

		expect(
			container.querySelector('.progress-bar[aria-valuenow="1"]')
		).not.toBeNull();
		expect(
			container.querySelector('.progress-bar__indicator')
		).not.toBeNull();
		expect(
			container.querySelector('.progress-bar__indicator').style.transform
		).toBe('translateX(-97%)');
	});
});
