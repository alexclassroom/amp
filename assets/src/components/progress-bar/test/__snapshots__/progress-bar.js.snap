// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar matches the snapshot 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={33}
  className="progress-bar"
  role="progressbar"
>
  <div
    className="progress-bar__track"
  >
    <div
      className="progress-bar__indicator"
      style={
        Object {
          "transform": "translateX(-67%)",
          "transitionDuration": "800ms",
        }
      }
    />
  </div>
</div>
`;
